mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -702,14 +702,8 @@ class UnresolvedMaterializationRewrite : public OperationRewrite {
     return rewrite->getKind() == Kind::UnresolvedMaterialization;
   }
 
-  UnrealizedConversionCastOp getOperation() const {
```

Comment:
I think it would be cleaner to keep this method for the code below, warn about the issue of potentially being erased and the need to check for it, and then add a method `bool erasedRewrite(OperationRewrite* rewrite)` or similar to `SingleEraseRewriter`. Since the static type is then `OperationRewrite` it'd be accessing the `Operation*`